package/src/netmon.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) CERN, 2018
+//
+// Network monitoring class
+//
+// This is a more efficient network monitor that holds
+// its istreams in a structure, so that they don't have
+// to be opened and closed on each monitoring cycle.
+
+#ifndef PRMON_NETMON_H
+#define PRMON_NETMON_H 1
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class netmon {
+ private:
+  // Which network interface paramters to measure
+  const std::vector<std::string> interface_params;
+
+  // Which network interfaces to monitor
+  std::vector<std::string> monitored_netdevs;
+
+  // Nested dictionary of network_if_streams[PARMETER][DEVICE][ISTREAM*]
+  std::unordered_map<
+      std::string,
+      std::unordered_map<std::string, std::unique_ptr<std::ifstream>>>
+      network_if_streams;
+
+  // Container for starting value stats, that are subtracted from measured
+  // values
+  std::unordered_map<std::string, unsigned long long> network_stats_start;
```

Comment:
`const` maybe? I don't think we're planning to change this throughout the program execution anyhow.

Reply:
But it has to be set to the correct values, so it can't be `const` here. Also, it may be that for error handling (like interfaces failing) we need to update it.

Reply:
OK, I see now that this was a dubious comment. I was thinking we could utilize the initialization list to do the assignment at run-time but I see we compute what we pass to this guy in the constructer.